### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by alexsuter**
- Enable deprecation warnings in build
- Use Java Compiler recorder intead of ECJ

### CODE REVIEW

File: src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java [hunk 1]
```diff
@@ -81,13 +81,9 @@ private Optional<MavenProject> findReactorProject(Artifact artifact) {
...
       return Optional.empty();
     }
-    var projects = session.getProjectMap();
-    if (projects == null) {
-      return Optional.empty();
-    }
-    String artifactKey = artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + artifact.getVersion();
-    MavenProject reactorProject = projects.get(artifactKey);
-    return Optional.ofNullable(reactorProject);
+    return session.getAllProjects().stream()
```
Comment (ivy-lmu) [inline src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java:84]: getAllProjects() cannot be null?

**Review by ivy-lmu — APPROVED** [review]

File: src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java (hunk 1 above)
Comment (alexsuter) [reply src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java:84]: I don't hope so. Otherwise I will add a null check :(

Comment (ivy-lmu) [reply src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java:84]: I'm not sure if this is the right place for hope.

Comment (alexsuter) [reply src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java:84]: God bless you.

**Pull request merged**